ProjectLibertyLabs/gateway
# fix: make all docs consistent with version prefix and grouping in sidebar

# Problem

Docs sidebar for different services/controllers was inconsistent; showing the version prefix path for some, and not others.

# Solution
Utilize Redocly `x-tagGroups` extension to group controllers with different versions nested under the controller group. Only affects docs.

## Steps to Verify:

1. Run `npm run generate:openapi`
2. Run `npm run generate:swagger-ui`
3. Inspect docs for each service to see changes
4. Run each service & browse to the live Swagger UI; should be no change from previous (Swagger UI doesn't support tag groups)

## Screenshots (optional):
![image](https://github.com/user-attachments/assets/59427902-a2b8-46d0-bdee-c8dfda0e579e)

![image](https://github.com/user-attachments/assets/fe16b5bb-e4fc-4f14-bd79-d5470bc5ac50)

![image](https://github.com/user-attachments/assets/9c7e6487-c3ca-4788-8981-02c705b017e6)

![image](https://github.com/user-attachments/assets/1c69e55e-21c6-406b-ba55-e37910545a33)


Closes #658
